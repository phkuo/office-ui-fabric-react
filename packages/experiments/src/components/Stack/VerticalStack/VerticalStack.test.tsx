import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import { VerticalStack } from '../index';

describe('VerticalStack', () => {
  it('can handle having no children', () => {
    const createEmptyStack = () => {
      mount(<VerticalStack />);
    };

    expect(createEmptyStack).not.toThrow();
  });

  it('renders default VerticalStack correctly', () => {
    const component = renderer.create(
      <VerticalStack>
        <div>Item 1</div>
        <div>Item 2</div>
      </VerticalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders gap correctly', () => {
    const component = renderer.create(
      <VerticalStack gap={5}>
        <div>Item 1</div>
        <div>Item 2</div>
      </VerticalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders growing items correctly', () => {
    const component = renderer.create(
      <VerticalStack>
        <VerticalStack.Item>Item 1</VerticalStack.Item>
        <VerticalStack.Item>Item 2</VerticalStack.Item>
      </VerticalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders shrinking items correctly', () => {
    const component = renderer.create(
      <VerticalStack shrinkItems>
        <VerticalStack.Item>Item 1</VerticalStack.Item>
        <VerticalStack.Item preventShrink>Item 2</VerticalStack.Item>
      </VerticalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders item alignments correctly', () => {
    const component = renderer.create(
      <VerticalStack>
        <VerticalStack.Item align="auto">Auto</VerticalStack.Item>
        <VerticalStack.Item align="stretch">Stretch</VerticalStack.Item>
        <VerticalStack.Item align="baseline">Baseline</VerticalStack.Item>
        <VerticalStack.Item align="start">Start</VerticalStack.Item>
        <VerticalStack.Item align="center">Center</VerticalStack.Item>
        <VerticalStack.Item align="end">End</VerticalStack.Item>
      </VerticalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
