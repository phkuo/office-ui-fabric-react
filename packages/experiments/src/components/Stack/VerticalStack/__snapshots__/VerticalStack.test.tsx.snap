// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalStack renders default VerticalStack correctly 1`] = `
<div
  className=
      ms-VerticalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        box-sizing: border-box;
        display: flex;
        flex-direction: column;
        flex-wrap: nowrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
        white-space: nowrap;
      }
      & > *:not(:empty) ~ * {
        margin-top: 0px;
      }
      & > *:empty {
        display: none;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 0;
      }
>
  <div>
    Item 1
  </div>
  <div>
    Item 2
  </div>
</div>
`;

exports[`VerticalStack renders gap correctly 1`] = `
<div
  className=
      ms-VerticalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        box-sizing: border-box;
        display: flex;
        flex-direction: column;
        flex-wrap: nowrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
        white-space: nowrap;
      }
      & > *:not(:empty) ~ * {
        margin-top: 5px;
      }
      & > *:empty {
        display: none;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 0;
      }
>
  <div>
    Item 1
  </div>
  <div>
    Item 2
  </div>
</div>
`;

exports[`VerticalStack renders growing items correctly 1`] = `
<div
  className=
      ms-VerticalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        box-sizing: border-box;
        display: flex;
        flex-direction: column;
        flex-wrap: nowrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
        white-space: nowrap;
      }
      & > *:not(:empty) ~ * {
        margin-top: 0px;
      }
      & > *:empty {
        display: none;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 0;
      }
>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Item 1
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Item 2
  </span>
</div>
`;

exports[`VerticalStack renders item alignments correctly 1`] = `
<div
  className=
      ms-VerticalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        box-sizing: border-box;
        display: flex;
        flex-direction: column;
        flex-wrap: nowrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
        white-space: nowrap;
      }
      & > *:not(:empty) ~ * {
        margin-top: 0px;
      }
      & > *:empty {
        display: none;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 0;
      }
>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: auto;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Auto
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: stretch;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Stretch
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: baseline;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Baseline
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: flex-start;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Start
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: center;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Center
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: flex-end;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    End
  </span>
</div>
`;

exports[`VerticalStack renders shrinking items correctly 1`] = `
<div
  className=
      ms-VerticalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        box-sizing: border-box;
        display: flex;
        flex-direction: column;
        flex-wrap: nowrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
        white-space: nowrap;
      }
      & > *:not(:empty) ~ * {
        margin-top: 0px;
      }
      & > *:empty {
        display: none;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 1;
      }
>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          flex-shrink: 1;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Item 1
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Item 2
  </span>
</div>
`;
